### DISCUSSION TITLE
How to share configuration with other devs

### DISCUSSION THREAD

**Discussion by tiagodj** [Escape-Technologies/mookme#93] [General]
So, I've setup hookme in my project and everything works beautifully. Kudos!

When my colleague pulled the changes, he didn't get the `.git/hooks/pre-commit` file created for him. I noticed that the only way to get that automated is if I run `npx mookme init --only-hook` and select `pre-commit`, but it will be a hard sell if I have to convince everyone in my team to do that.

how do I get them to have the pre-commit configured? 

I was hoping that when they do `yarn install` in the project root folder, it would create it.
I could add a package.json script that does that, if I could specify which hooks, for example:

`npx mookme init --only-hook pre-commit`

Or am I missing something?

We don't add the `.git` folder and its files to the repository.

**Comment by LMaxence** [comment]
Hello !

First of all, many thanks for trying my tool :) 

I totally agree with you regarding this configuration problem. The problem is indeed that the `.git` folder is not versioned, and, theoretically, each developer is supposed to defined his own git hooks.

Yet, what we currently do at my company is that we run `npx mookme init --only-hook --skip-types-selection`.

This will run the initialization, with no prompters at all, and make every type of git hook call Mookme. I realize that this option is not documented, and that we should allow to provide a list of git hooks as well (I will open an issue regarding this)

Thus, we can "enforce" (I don't really like this I think that talking people into using it is more efficient) mookme's usage at the moment developer setup their environment.

Regarding postinstall scripts, I'd rather not rely on them, since they are not called by every package managers
